File: springboot-java-backend/src/main/java/com/example/template/routes/PingController.java
package com.example.template.routes;

import com.example.template.types.SuccessResponse;
import io.swagger.v3.oas.annotations.Operation;
import java.time.Instant;
import java.util.Map;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1")
public class PingController {
  @Operation(summary = "Ping endpoint")
  @GetMapping("/ping")
  public SuccessResponse<Map<String, Object>> ping() {
    return SuccessResponse.of(Map.of("message", "pong", "timestamp", Instant.now().toString()));
  }

  @GetMapping("/fail")
  public SuccessResponse<Map<String, Object>> fail() {
    throw new IllegalStateException("Boom");
  }
}

File: laravel-php-backend/app/Http/Controllers/PingController.php
<?php

declare(strict_types=1);

namespace App\Http\Controllers;

use App\Types\Api;
use RuntimeException;

final class PingController
{
    public function ping(): array
    {
        if (isset($_GET['fail']) && $_GET['fail'] === '1') {
            throw new RuntimeException('Forced ping failure');
        }

        return Api::success(['message' => 'pong']);
    }
}
